### ISSUE TITLE
Wrong result when enclosing `EXISTS` in a `CASE` expression

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13119]
I found a discrepancy when running two semantically equivalent queries against an empty database:

---
1:
```cypher
CREATE ({x:EXISTS {  MATCH () } })
-
MATCH ({x:true}) RETURN 0
```
Returns no records.

---
2:
```cypher
CREATE ({x:CASE WHEN true THEN EXISTS {  MATCH () } END})
-
MATCH ({x:true}) RETURN 0
```
Produces:
```
╒═══╕
│"0"│
╞═══╡
│0  │
└───┘
```
---

I believe the queries mentioned above are semantically equivalent and thus their result should be the same.

I encountered this issue when testing queries against the **Neo4j 5.6.0 community version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the aforementioned, semantically equivalent, queries against an empty database and observe that their results differ.

### Expected behavior
The queries should result in the same behavior

### Actual behavior
The first query returns an empty result, whereas the second query doesn't.

**Comment by AzuObs** [comment]
Hi Dominic, 

Thank you for reporting the issue. I also think there is something strange, or at least not obvious, happening here. I have submitted the issue ([private ticket](https://trello.com/c/SKfxtvct)) to the appropriate team and hope they will get back to you shortly.

All the best,

**Comment by sherfert** [comment]
Hi @DominicWuest 

You will be delighted to hear that we finally got around to fixing this bug. The fix will be part of the upcoming 5.17 release.
